### ISSUE TITLE
[HOLD for payment 2023-11-22] [$500]  New room - Workspace selector doesn't highlight when selected

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#29675]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.84.1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @namhihi237
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1697209356639479

## Action Performed:
1. Create some WS
2. Click FAB button
3. Click send message and select room tab
4. Click workspace section
5. Select any workspace
6. Click workspace section again, Observe that the selected item is not highlighted

## Expected Result:
Selected item is highlighted

## Actual Result:
Selected item is not highlighted

## Workaround:
Unknown 

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

 

https://github.com/Expensify/App/assets/93399543/bdcfc8d8-c169-44eb-94d0-489f6af7aad4

</details>

<details>
<summary>Android: mWeb Chrome</summary>

 

https://github.com/Expensify/App/assets/93399543/99c9d69a-ff21-4ff0-a043-f056ae585698

</details>

<details>
<summary>iOS: Native</summary>

 

https://github.com/Expensify/App/assets/93399543/f434e186-f572-426a-b1a1-b16af76f024c

</details>

<details>
<summary>iOS: mWeb Safari</summary>

 

https://github.com/Expensify/App/assets/93399543/959c275c-e7b1-4a90-9e5e-7fa2af95db88

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

 

https://github.com/Expensify/App/assets/93399543/ce3b4371-63e0-4fa6-8846-7b8a19729018

</details>

<details>
<summary>MacOS: Desktop</summary>

 

https://github.com/Expensify/App/assets/93399543/0abf51cc-b45c-48db-8730-7080aaff528e

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a505b5f852bd2d0f</li>
        <li>Upwork Job ID: 1713907491046084608</li>
        <li>Last Price Increase: 2023-10-30</li>
<li>Automatic offers: </li>
<ul>
<li>0xmiroslav | Reviewer | 27552192</li>
<li>namhihi237 | Contributor | 27552193</li>
<li>namhihi237 | Reporter | 27552194</li>
</ul></li>
    </ul>
</details>

**Comment by kbecciv** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
The selected item is highlighted on some selectors on Create Room, we need to fix them
### What is the root cause of that problem?
In Visibility and Workspace selectors when creating a room we use `ValueSelectorModal` component and it uses `SelectionList` but we pass the `initiallyFocusedOptionKey` incorrectly.
https://github.com/Expensify/App/blob/9e19e40aef0f303ab5d5fc1c9cf8c9caf86a8098/src/components/ValuePicker/ValueSelectorModal.js#L70-L74
It should be the same we create keyForList in `itemsData`
https://github.com/Expensify/App/blob/9e19e40aef0f303ab5d5fc1c9cf8c9caf86a8098/src/components/ValuePicker/ValueSelectorModal.js#L47
### What changes do you think we should make in order to solve the problem?
We should update correctly `initiallyFocusedOptionKey`:
```js
initiallyFocusedOptionKey={selectedItem.value}
```
### What alternative solutions did you explore? (Optional)
Also, we can remove `currentValue` prop because it doesn't used

**Comment by joekaufmanexpensify** [comment]
Going to triage this tomorrow

**Comment by joekaufmanexpensify** [comment]
Seems like this is a bug, but confirming [here](https://expensify.slack.com/archives/C049HHMV9SM/p1697836137452579?thread_ts=1697209356.639479&cid=C049HHMV9SM) that this should not have been fixed in another issue

**Comment by joekaufmanexpensify** [comment]
Okay, confirmed we can proceed here. @0xmiroslav could you share whether any of these proposals work?

**Comment by joekaufmanexpensify** [comment]
Pending review of proposals

**Comment by joekaufmanexpensify** [comment]
TY!

**Comment by joekaufmanexpensify** [comment]
@0xmiroslav any of these proposals work?

**Comment by joekaufmanexpensify** [comment]
Still pending input on proposals

**Comment by joekaufmanexpensify** [comment]
Pending sign off from @dangrous

**Comment by dangrous** [comment]
Hi, I will look through the discussion here but first - I'm a little confused as to what the bug is. What SHOULD the selected item look like? Right now it is bold and has a checkmark. Is there something else missing?

**Comment by dangrous** [comment]
ah okay with the highlight - thank you!

So I think I agree with @0xmiroslav in this case - although @dukenv0307 is not wrong that the coding of SelectionList is a bit weird - I think it's safer to focus on updating this particular case, and that way we won't have to worry as much about regressions. At some point, I think it would be worth taking another look at updating the logic the way they mentioned, but I think for now let's move forward with @namhihi237's thoughts.

Thank you for the detailed discussion!

**Comment by dangrous** [comment]
@namhihi237 do you mind commenting on this issue so I can assign you? Thanks!

**Comment by joekaufmanexpensify** [comment]
Not overdue

**Comment by dangrous** [comment]
I'm going to assign @namhihi237 in this case but thank you for the discussion @dukenv0307 ! I really appreciate your thoughts and persistence. We can definitely revisit if we end up needing to in the future.

**Comment by joekaufmanexpensify** [comment]
@0xmiroslav I saw your comment on the other issue. That means the PR in this issue is blocked on #31192, is that right?

**Comment by joekaufmanexpensify** [comment]
@0xmiroslav could you please complete your portion of the BZ checklist?

**Comment by joekaufmanexpensify** [comment]
BZ checklist is done. All set to issue payment!

**Comment by joekaufmanexpensify** [comment]
We need to issue the following payments: 

- @namhihi237 - $500 for PR. Paid via Upwork.
- @namhihi237 - $50 for reporting. Paid via Upwork. 
- @0xmiroslav -$500 for C+ review.

**Comment by joekaufmanexpensify** [comment]
@namhihi237 $500 sent and contract ended!

**Comment by joekaufmanexpensify** [comment]
@namhihi237 $50 sent and contract ended!

**Comment by joekaufmanexpensify** [comment]
Sounds good! There was an automatic offer sent to an upwork account for a Volodymyr for C+ review. Going to end that contract without payment for now.
